daemon/update.go

```diff
@@ -74,11 +73,6 @@ func (daemon *Daemon) update(name string, hostConfig *container.HostConfig) erro
 	// if Restart Policy changed, we need to update container monitor
 	container.UpdateMonitor(hostConfig.RestartPolicy)
 
-	// if container is restarting, wait 5 seconds until it's running
-	if container.IsRestarting() {
```

Comment:
Actually this piece of code was first introduced by me, I was hoping to let container reach a stable state(running/stopped) before trying to update the container's resources. After some consideration, I think even we add polling (say 5 seconds), it can't guarantee the container can escape from the restarting state, maybe it can decrease the risk of race but can't remove the risk all. 

So I think it's safe to remove this, and save the blocking time then user can get a quicker feedback.

What do you think?